lib/a15k/exporter.rb

```diff
@@ -0,0 +1,90 @@
+require 'a15k/html_preview'
+
+module A15k
+  class Exporter
+
+    class CreateAssessmentError < StandardError; end
+
+    def initialize
+      @outcomes = {
+        success_count: 0,
+        failure_info: []
+      }
+    end
+
+    def run
+      format = make_sure_format_is_uploaded_and_return
+
+      Exercise.published                     # Select only published exercises
+              .can_release_to_a15k           # That have been marked as releaseable to a15k
+              .not_released_to_a15k          # And that haven't yet been released
+              .find_each do |exercise|
+
+        begin
+          export_one_exercise(exercise, format)
+
+          @outcomes[:success_count] += 1
+        rescue CreateAssessmentError, A15kClient::ApiError => ee
+          @outcomes[:failure_info].push({uid: exercise.uid, message: ee.message})
+        end
+
+      end
+
+      @outcomes
+    end
+
+    def make_sure_format_is_uploaded_and_return
+      # See if it is already uploaded
+      format = formats_api.get_formats
+                          .data
+                          .find{|format| format.identifier == local_format_data['identifier']}
+
+      # If it is uploaded, return it; otherwise upload it via the API and return it
+      format || formats_api.create_format(local_format_data).data
+    end
+
+    def local_format_data
+      @local_format_data ||= YAML.load_file Rails.root.join('lib/a15k', 'format.yml')
+    end
+
+    def export_one_exercise(exercise, format)
+      # Get the exercise JSON; we toss out "community solutions" for licensing
+      # reasons.
+
+      exercise_data = HashWithIndifferentAccess.new(Api::V1::Exercises::Representer.new(exercise).to_hash)
+      exercise_data[:questions].each do |question_data|
```

Comment:
do you see this file being used as an example of writing an uploader?  If so why is the `HashWithIndifferentAccess` needed?  

I suspect you're using it because somewhere in the a15k lib it uses both string and symbol access. If that's the case we should document why we need it with a comment or something